### ISSUE TITLE
Extract sync logic from main.rs

### ISSUE THREAD

**Issue by wlthomson** [openmsupply/remote-server#234]
From epic: https://github.com/openmsupply/remote-server/issues/236

`main.rs` includes a proof-of-concept for message-based sync and scheduling (props to @chris-morgan). The idea is to expand this example out into a minimal synchronisation system with a sync actor for "pulling" and "pushing" data (`SyncActor`) and other actors responsible for sending messages to trigger sync operations (e.g. `SyncScheduler`). 

This task is probably too big for a single issue, so the first step is to extract the existing logic out of `main.rs` and into the `remote-server` library. Once this is done, it should be easier to identify what needs to be refactored and what functionality needs to be added to implement basic sync functionality.

**Comment by wlthomson** [comment]
Duplicate of #248.
